src/main/java/uk/gov/ons/ssdc/supporttool/security/UserIdentity.java

```diff
@@ -1,154 +1,37 @@
 package uk.gov.ons.ssdc.supporttool.security;
 
-import com.godaddy.logging.Logger;
-import com.godaddy.logging.LoggerFactory;
-import com.google.api.client.json.webtoken.JsonWebToken;
 import com.google.auth.oauth2.TokenVerifier;
-import java.util.Optional;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
-import org.springframework.util.StringUtils;
-import org.springframework.web.server.ResponseStatusException;
 import uk.gov.ons.ssdc.common.model.entity.Survey;
-import uk.gov.ons.ssdc.common.model.entity.User;
 import uk.gov.ons.ssdc.common.model.entity.UserGroupAuthorisedActivityType;
-import uk.gov.ons.ssdc.common.model.entity.UserGroupMember;
-import uk.gov.ons.ssdc.common.model.entity.UserGroupPermission;
-import uk.gov.ons.ssdc.supporttool.model.repository.UserRepository;
 
 @Component
 public class UserIdentity {
-  private static final Logger log = LoggerFactory.getLogger(UserIdentity.class);
   private static final String IAP_ISSUER_URL = "https://cloud.google.com/iap";
-
-  private final UserRepository userRepository;
   private final String iapAudience;
-  private final boolean dummyUserIdentityAllowed;
-  private final String dummyUserIdentity;
-  private final String dummySuperUserIdentity;
+
+  private final AuthUser authUser;
 
   private TokenVerifier tokenVerifier = null;
 
-  public UserIdentity(
-      UserRepository userRepository,
-      @Value("${iapaudience}") String iapAudience,
-      @Value("${dummyuseridentity-allowed}") boolean dummyUserIdentityAllowed,
-      @Value("${dummyuseridentity}") String dummyUserIdentity,
-      @Value("${dummysuperuseridentity}") String dummySuperUserIdentity) {
-    this.userRepository = userRepository;
+  public UserIdentity(AuthUser authUser, @Value("${iapaudience}") String iapAudience) {
```

Comment:
Does this class need to exist anymore? It looks like with the refactor it is just passing all function calls through, could it be removed, and the new interface called directly instead?

Reply:
good spot, it doesn't need to exist anymore. I've removed it but the thing I'm not sure on is the synchronised method `getTokenVerifier()` for when IAP is disabled. Since I've read Components are Singletons so there should only be on instance of DummyUser and then therefor one instance of IAPUser stored in that singleton, but then not 100% sure if synchronised will work as intended?